packages/api/src/clickhouse/index.ts

```diff
@@ -1017,6 +1022,78 @@ export const buildMetricSeriesQuery = async ({
     ],
   );
 
+  const quantile =
+    aggFn === AggFn.P50
+      ? '0.5'
+      : aggFn === AggFn.P90
+      ? '0.90'
+      : aggFn === AggFn.P95
+      ? '0.95'
+      : '0.99';
+
+  const histogramMetricSource = SqlString.format(
+    `
+      WITH points AS (
+        SELECT
+          toStartOfInterval(timestamp, INTERVAL ?) AS timestamp,
+          name,
+          arraySort((x) -> x[2], groupArray([
+            toUInt64(value),
+            toUInt64OrDefault(_string_attributes['le'], 18446744073709551615)
+          ])) AS point,
+          mapFilter((k, v) -> (k != 'le'), _string_attributes) AS filtered_string_attributes
+        FROM ??
+        WHERE name = ?
+        AND data_type = ?
+        AND (?)
+        AND mapContains(_string_attributes, 'le')
+        GROUP BY timestamp, name, filtered_string_attributes
+      ), diff_points AS (
+        SELECT 
+          timestamp,
+          name,
+          filtered_string_attributes AS _string_attributes,
+          arrayMap((x) -> x[1], point) AS _points,
+          arrayMap((x) -> x[2], point) AS buckets,
+          (_points - arrayShiftRight(_points, 1)) AS deltas
+        FROM points
+      ), flatten_points AS (
+        SELECT 
+          timestamp,
+          name,
+          _string_attributes,
+          (arrayJoin(arrayZip(deltas, buckets)) AS t).1 AS delta,
+          t.2 AS bucket
+        FROM diff_points
+      )
+      SELECT
+        timestamp,
+        name,
+        _string_attributes,
+        quantileTimingWeighted(toFloat32(?))
```

Comment:
Now that we have this working, we might want to use `quantileInterpolatedWeighted` - should be a drop in replacement, and will give a better approximation for the actual value of the quantile 👍